packages/app/src/core/models/PluginAttributesManager/PluginAttributesManager.ts

```
@@ -11,78 +11,215 @@ const PluginAttributesManager = types
+
     setStateRequest: types.optional(RequestModel, {}),
+    deleteStateRequest: types.optional(RequestModel, {}),
     getConfigRequest: types.optional(RequestModel, {})
   })
   .actions((self) => ({
-    get: flow(function* <T>(spaceId: string, key: string) {
-      const response = yield self.getStateRequest.send(
-        api.spaceAttributeRepository.getSpaceSubAttribute,
+    getSpaceAttributeValue: flow(function* <T extends AttributeValueInterface>(
+      spaceId: string,
+      attributeName: string
+    ) {
+      const response = yield self.getAttributeRequest.send(
+        api.spaceAttributeRepository.getSpaceAttribute,
+        {
+          spaceId,
+          plugin_id: self.pluginId,
+          attribute_name: attributeName
+        }
+      );
+
+      if (response === undefined) {
+        return {} as T;
+      }
+
+      return response as T;
+    }),
+    setSpaceAttributeValue: flow(function* <T extends AttributeValueInterface>(
+      spaceId: string,
+      attributeName: string,
+      value: T
+    ) {
+      const response = yield self.getAttributeRequest.send(
+        api.spaceAttributeRepository.setSpaceAttribute,
+        {
+          spaceId,
+          plugin_id: self.pluginId,
+          attribute_name: attributeName,
+          value
+        }
+      );
+
+      if (response === undefined) {
+        throw Error('Empty response');
+      }
+
+      return response as T;
+    }),
+    deleteSpaceAttribute: flow(function* (spaceId: string, attributeName: string) {
+      const response = yield self.deleteAttributeRequest.send(
+        api.spaceAttributeRepository.deleteSpaceAttribute,
         {
           spaceId,
           plugin_id: self.pluginId,
-          attribute_name: AttributeNameEnum.STATE,
+          attribute_name: attributeName
+        }
+      );
+
+      if (response === undefined) {
+        throw Error('Empty response');
+      }
+
+      return response;
```

Btw: delete action returns nothing usually.

---

returns null if successfully deleted

---

I think `send()` should throw an exception in case of error but it's a pretty big change and subject to bigger refactoring.